File: get_playlist.js
const YTMusic = require('ytmusicapi').default;
const fs = require('fs');

async function main() {
    // 1. Load your OAuth JSON from the environment variable
    const oauthRaw = process.env.YTM_OAUTH_JSON;
    if (!oauthRaw) {
        console.error("Error: YTM_OAUTH_JSON secret is missing!");
        process.exit(1);
    }

    // 2. Initialize the client
    // We write the file because the library expects a path
    fs.writeFileSync('oauth.json', oauthRaw);
    const ytmusic = new YTMusic('oauth.json');

    const playlistId = "PL8WGYt2fhenCJnBHFBKqw8SZl-oyO03Ur";
    const targetVideoId = "-oIMyMEvMLI"; // The one you want to remove

    console.log(`Accessing Private Playlist...`);

    try {
        // 3. Fetch Playlist
        const playlist = await ytmusic.getPlaylist(playlistId);
        console.log(`--- SUCCESS! Found: ${playlist.title} ---`);

        // 4. Find and Remove the item
        const trackToRemove = playlist.tracks.find(t => t.videoId === targetVideoId);

        if (trackToRemove) {
            console.log(`Found: ${trackToRemove.title}. Attempting removal...`);
            
            // In JS, we pass the playlistId and the track's setVideoId
            const response = await ytmusic.removePlaylistItems(playlistId, [trackToRemove]);
            console.log("Removal Status:", response);
        } else {
            console.log("Track not found in playlist. It might already be gone.");
        }

    } catch (error) {
        console.error("Operation Failed!");
        console.error("Server Message:", error.message);
        
        if (error.message.includes("400")) {
            console.log("\nSTILL GETTING 400? CHECK THESE:");
            console.log("1. Ensure your Client ID is 'TVs and Limited Input Devices'.");
            console.log("2. Verify you CHECKED THE BOX for 'Manage YouTube' during login.");
        }
    }
}

main();

File: get_playlist.py
import os
import json
from ytmusicapi import YTMusic, OAuthCredentials

def get_ytm_client():
    oauth_raw = os.environ.get("YTM_OAUTH_JSON")
    client_id = os.environ.get("YTM_CLIENT_ID")
    client_secret = os.environ.get("YTM_CLIENT_SECRET")

    # Write the file for the library
    with open("oauth.json", "w") as f:
        f.write(oauth_raw)

    auth = OAuthCredentials(
        client_id=client_id,
        client_secret=client_secret
    )

    # Note: If your account has multiple 'Channels', 
    # you might need to specify the brand_id, but usually it's None.
    return YTMusic("oauth.json", oauth_credentials=auth)

def main():
    yt = get_ytm_client()
    
    # Let's try to get the library playlists first to see if auth is valid
    print("Testing connection...")
    try:
        # If this works, your OAuth is 100% fine
        test = yt.get_library_playlists(limit=1)
        print("Auth check: Success!")
    except Exception as e:
        print(f"Auth check failed: {e}")

    # The Playlist ID
    playlist_id = "PL8WGYt2fhenCJnBHFBKqw8SZl-oyO03Ur"
    
    try:
        # Use a small limit and no related data to simplify the request
        playlist = yt.get_playlist(playlist_id, limit=50)
        
        print(f"Playlist Title: {playlist.get('title', 'Unknown')}")
        
        # Save tracks to a file for your other Reels scripts
        tracks_data = []
        for track in playlist.get('tracks', []):
            info = {
                "id": track['videoId'],
                "title": track['title'],
                "artist": track['artists'][0]['name'] if track['artists'] else "Unknown"
            }
            tracks_data.append(info)
            print(f"Added: {info['artist']} - {info['title']}")

        with open("playlist_items.json", "w") as f:
            json.dump(tracks_data, f, indent=4)
            
    except Exception as e:
        print(f"Error fetching playlist: {e}")

if __name__ == "__main__":
    main()
